Flaky test in server registration endpoint TestMintJWTSVID/success_with_default_TTL

I bumped into this in master, details below. Expected time off by one second.

```
--- FAIL: TestHandler (0.21s)
    --- FAIL: TestHandler/TestMintJWTSVID (0.01s)
        --- FAIL: TestHandler/TestMintJWTSVID/success_with_default_TTL (0.00s)
            require.go:157: 
                	Error Trace:	handler_test.go:1063
                	Error:      	Not equal: 
                	            	expected: time.Time{wall:0x0, ext:63701937640, loc:(*time.Location)(nil)}
                	            	actual  : time.Time{wall:0x0, ext:63701937639, loc:(*time.Location)(nil)}
                	            	
                	            	Diff:
                	            	--- Expected
                	            	+++ Actual
                	            	@@ -1,2 +1,2 @@
                	            	-(time.Time) 2019-08-20 22:40:40 +0000 UTC
                	            	+(time.Time) 2019-08-20 22:40:39 +0000 UTC
                	            	 
                	Test:       	TestHandler/TestMintJWTSVID/success_with_default_TTL
FAIL
coverage: 89.6% of statements
FAIL	github.com/spiffe/spire/pkg/server/endpoints/registration	0.287s
```

--- a/pkg/agent/manager/manager_test.go
+++ b/pkg/agent/manager/manager_test.go
@@ -55,7 +55,7 @@ var (
 )
 
 func TestInitializationFailure(t *testing.T) {
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	ca, cakey := createCA(t, clk, trustDomain)
 	baseSVID, baseSVIDKey := createSVID(t, clk, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
 	cat := fakeagentcatalog.New()
@@ -85,7 +85,7 @@ func TestStoreBundleOnStartup(t *testing.T) {
 	dir := createTempDir(t)
 	defer removeTempDir(dir)
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	ca, cakey := createCA(t, clk, trustDomain)
 	baseSVID, baseSVIDKey := createSVID(t, clk, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
 	cat := fakeagentcatalog.New()
@@ -137,7 +137,7 @@ func TestStoreSVIDOnStartup(t *testing.T) {
 	dir := createTempDir(t)
 	defer removeTempDir(dir)
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	ca, cakey := createCA(t, clk, trustDomain)
 	baseSVID, baseSVIDKey := createSVID(t, clk, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
 	cat := fakeagentcatalog.New()
@@ -185,7 +185,7 @@ func TestStoreKeyOnStartup(t *testing.T) {
 	dir := createTempDir(t)
 	defer removeTempDir(dir)
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	ca, cakey := createCA(t, clk, trustDomain)
 	baseSVID, baseSVIDKey := createSVID(t, clk, ca, cakey, "spiffe://"+trustDomain+"/agent", 1*time.Hour)
 
@@ -252,7 +252,7 @@ func TestHappyPathWithoutSyncNorRotation(t *testing.T) {
 	}
 	defer l.Close()
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
 		t:             t,
 		trustDomain:   trustDomain,
@@ -590,7 +590,7 @@ func TestSynchronizationClearsStaleCacheEntries(t *testing.T) {
 	}
 	defer l.Close()
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
 		t:             t,
 		trustDomain:   trustDomain,
@@ -652,7 +652,7 @@ func TestSynchronizationUpdatesRegistrationEntries(t *testing.T) {
 	}
 	defer l.Close()
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
 		t:             t,
 		trustDomain:   trustDomain,
@@ -713,7 +713,7 @@ func TestSubscribersGetUpToDateBundle(t *testing.T) {
 	}
 	defer l.Close()
 
-	clk := clock.New()
+	clk := clock.NewMock(t)
 	apiHandler := newMockNodeAPIHandler(&mockNodeAPIHandlerConfig{
 		t:             t,
 		trustDomain:   trustDomain,

--- a/pkg/agent/plugin/workloadattestor/docker/retry.go
+++ b/pkg/agent/plugin/workloadattestor/docker/retry.go
@@ -5,7 +5,7 @@ import (
 	"math"
 	"time"
 
-	"github.com/spiffe/spire/test/clock"
+	"github.com/andres-erbsen/clock"
 )
 
 const (

--- a/pkg/server/ca/ca.go
+++ b/pkg/server/ca/ca.go
@@ -344,9 +344,9 @@ func (ca *CA) SignJWTSVID(ctx context.Context, params JWTSVIDParams) (string, er
 
 	telemetry_server.IncrServerCASignJWTSVIDCounter(ca.c.Metrics, params.SpiffeID)
 	ca.c.Log.WithFields(logrus.Fields{
-		telemetry.Audience: params.Audience,
+		telemetry.Audience:   params.Audience,
 		telemetry.Expiration: expiresAt.Format(time.RFC3339),
-		telemetry.SPIFFEID: params.SpiffeID,
+		telemetry.SPIFFEID:   params.SpiffeID,
 	}).Debug("Server CA successfully signed JWT SVID")
 
 	return token, nil

--- a/pkg/server/plugin/datastore/sql/sql_test.go
+++ b/pkg/server/plugin/datastore/sql/sql_test.go
@@ -65,7 +65,7 @@ type PluginSuite struct {
 }
 
 func (s *PluginSuite) SetupSuite() {
-	clk := clock.New()
+	clk := clock.NewMock(s.T())
 
 	expiredNotAfterTime, err := time.Parse(time.RFC3339, _expiredNotAfterString)
 	s.Require().NoError(err)

--- a/test/clock/clock.go
+++ b/test/clock/clock.go
@@ -11,11 +11,6 @@ import (
 // Clock is a clock
 type Clock clock.Clock
 
-// New returns a Clock backed by a realtime clock
-func New() Clock {
-	return clock.New()
-}
-
 // Mock is a mock clock that can be precisely controlled
 type Mock struct {
 	*clock.Mock

--- a/test/fakes/fakeserverca/serverca.go
+++ b/test/fakes/fakeserverca/serverca.go
@@ -47,7 +47,7 @@ func New(t *testing.T, trustDomain string, options *Options) *CA {
 		options = new(Options)
 	}
 	if options.Clock == nil {
-		options.Clock = clock.New()
+		options.Clock = clock.NewMock(t)
 	}
 	if options.X509SVIDTTL == 0 {
 		options.X509SVIDTTL = time.Minute

